### ISSUE TITLE
About the Reinforce algorithm of the RAM model

### ISSUE THREAD

**Issue by vyouman** [Element-Research/rnn#175]
Hi, I'm reading the code of the RAM model you implemented in the rnn package example. And I'm kind of confused of the loss function of the model.

> loss = nn.ParallelCriterion(true)
> :add(nn.ModuleCriterion(nn.ClassNLLCriterion(), nil, nn.Convert())) -- BACKPROP
> :add(nn.ModuleCriterion(nn.VRClassReward(agent, opt.rewardScale), nil, nn.Convert())) -- REINFORCE

In the paper http://papers.nips.cc/paper/5542-recurrent-models-of-visual-attention.pdf, see the formula at the bottom of page 4, the author mentioned M samples si,

> si’s are interaction sequences obtained by running the current agent πθ for i = 1 . . . M
> episodes.

So where is the M variable present in the previous loss function? 
Many thanks.

**Comment by nicholas-leonard** [comment]
M is the number of iterations in the epoch, i.e. the number of samples in the dataset. This is handled by the sizeAverage attribute of most criterions in nn.

**Comment by vyouman** [comment]
@nicholas-leonard Oh, thanks a lot. It helps. :p

**Comment by vyouman** [comment]
@nicholas-leonard To be clear, I think M is the number of samples per batch here, right?

**Comment by nicholas-leonard** [comment]
Yes.

Nicholas Léonard
917-741-7570

**Comment by vyouman** [comment]
Thanks a lot. Let's close the issue. :p
